### ISSUE TITLE
Make discharge data from caravan available through the observations routine (as well as part of the forcing !)

### ISSUE THREAD

**Issue by RolfHut** [eWaterCycle/ewatercycle#462]
The caravan dataset contains both data we consider "forcing" (Temperature, precipitation, potential evap, etc.) as well as discharge data we consider "observations". Currently we make this data accesable through 'forcing'.

For discharge data it would be great if we can load caravan discharge data similar to how we load GRPC data (and this prepares us for adding eStreams at a later stage). It is no problem if "under the hood" this would just call the forcing part and only returns the discharge part of the caravan forcing.

**Comment by RolfHut** [comment]
addition: caravan is currenlty presented as a "forcing source", which we had reserved for the different types of forcing that models could intake. Conceptually it is much more a "dataset", like ERA5 and CMIP are. I suggest to rewrite generic lumped forcing to take Dataset = "caravan" as an argument (with additional required "caravan ID") and remove caravanforcing as a seperate source.

**Comment by BSchilperoort** [comment]
> addition: caravan is currenlty presented as a "forcing source", which we had reserved for the different types of forcing that models could intake. Conceptually it is much more a "dataset", like ERA5 and CMIP are. 

I disagree with this. In my view Caravan closer to "_already prepared_ forcing" unlike ERA5 or CMIP. ERA5 and CMIP are generic weather/climate model outputs. These won't directly fit into any hydrological models, but some steps are usually needed (e.g. aggregation to a polygon for lumped models). However, you can use them for any location on the globe.

The Caravan data is already preprocessed and aggregated in some way to allow for easy use in (catchment lumped) hydrological modelling. It is already catchment specific.

The "dataset" argument in the forcing classes refers to more generic (CMORized) data. These datasets should be compatible with all/most model-specific forcing generators (provided that they have the right variables). Other datasets in this context could be ERA5-land or MERRA-2.

**Comment by BSchilperoort** [comment]
> For discharge data it would be great if we can load caravan discharge data similar to how we load GRPC data (and this prepares us for adding eStreams at a later stage).

It would be nice if the observations module for eWaterCycle would be more standardized and allow for easier extension/addition. In the refactor for (py) ewatercycle 2.0 we focused on making the forcing and models more modular, but did not have time to tackle observations as well.

**Comment by BSchilperoort** [comment]
> Dataset = "caravan" as an argument (with additional required "caravan ID")

Also Caravan now (as of v1.5) has ERA5-land data with custom Penman Monteith based potential evaporation. There is also "Caravan MultiMet" which has multiple different source datasets to choose from for the forcing data.

**Comment by RolfHut** [comment]
ok, I'm convinced by your arguments. Let's keep "caravan" as a source at the same level as "genericLumped"
